lib/tasks/cleanup.rake

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+namespace :dpul do
+  namespace :cleanup do
+    desc "delete duplicate sidecars"
+    task sidecars: :environment do
+      config = CatalogController.blacklight_config
+      repo = FriendlyIdRepository.new(config)
+      dups = Spotlight::SolrDocumentSidecar.select(:exhibit_id, :document_id).group(:exhibit_id, :document_id).having("count(*) > 1").size.keys
+
+      dup_sets = dups.map do |exhibit_id, document_id|
+        Spotlight::SolrDocumentSidecar.where(exhibit_id: exhibit_id, document_id: document_id).to_a
+      end
+
+      dup_sets.each do |ds|
+        # for each set, if any sidecar has resource_id nil, delete that sidecar
+        linked_sidecars = ds.reject { |sidecar| sidecar.resource_id.nil? }
+        ds.select { |sidecar| sidecar.resource_id.nil? }.each do |sidecar|
+          puts "deleting sidecar #{sidecar.id}"
+          # sidecar.destroy
+        end
+
+        # otherwise, get the solr document and see which resource_id it references
+        next unless linked_sidecars.count > 1
+
+        doc = repo.search(q: "id:#{linked_sidecars.first.document_id}")["response"]["docs"].first
```

Comment:
Won't this always get the document for the given sidecar, which means it'll always match the resource ID of whatever the first sidecar is? Is there only one document, despite there being two resource IDs?

Reply:
Yes. There is only one document.

Reply:
Gotcha. So theoretically the document ID some combo of the exhibit ID and something else which is duplicated between these two sidecars.

Reply:
sidecars are only considered duplicate if they have the same exhibit id and the same document id -- this is what makes them error on the new uniqueness requirement introduced by the upgrade. So the duplicate sidecars have the same document but different resource ids. we figure whichever resource the document references is the real one.